### PR TITLE
Misc. improvements to JIT logging pathlength

### PR DESCRIPTION
**Pull request by 0xdaryl**
* Introduce appends() function to TR_PrettyPrinterString class 
* Use TR_PrettyPrinterString::appends() where appropriate 
* Rename TR_PrettyPrinterString::append() to appendf() 
* Remove log printing for obsolete symRefWCodeId 
* Remove excessive TR_PrettyPrinterString usage for globalIndexPrefix 
* Reduce calls to TR_PrettyPrinterString.appends() 
* Improve pathlength in TR_Debug::newNode() 
* Improve pathlength in TR_Debug::newLabelSymbol() 
* Improve pathlength in TR_Debug::newRegister() 
* Improve pathlength in TR_Debug::newVariableSizeSymbol() 
* Common break or debug on create functionality in TR_Debug

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Jenkins build all

**Comment by 0xdaryl** [comment]
Jenkins build win

**Comment by 0xdaryl** [comment]
Jenkins build all

**Comment by 0xdaryl** [comment]
Jenkins build win

Windows CI failure is infrastructural.

**Comment by 0xdaryl** [comment]
Much of this work was completed prior to my discovery of the `TR_StringBuf` class [1] which does largely the same thing as `TR_PrettyPrinterString`.  A follow-on, near-term goal is to have the RAS code use `TR_StringBuf` instead.  In the meantime, @dsouzai would you mind reviewing this please?

[1] https://github.com/eclipse/omr/blob/4529b9fe54bb89a69dbdc266d258c1716bd5b538/compiler/infra/String.hpp#L58

File: compiler/ras/Tree.cpp [hunk 1]
```diff
@@ -997,6 +986,26 @@ void TR_PrettyPrinterString::append(const char* format, ...)
 
    }
 
+void TR_PrettyPrinterString::appends(char const *str)
+   {
+   size_t const strLen0 = strlen(str) + 1; // include terminating '\0'
+   size_t const bufLenBytes = 2000-len;
```
Comment (dsouzai) [inline compiler/ras/Tree.cpp:992]: The 2000 should be made into a const static in the `TR_PrettyPrinterString` class.

**Review by dsouzai — COMMENTED** [review]
Looks OK to me; minor change requested.

File: compiler/ras/Tree.cpp (hunk 1 above)
Comment (0xdaryl) [reply compiler/ras/Tree.cpp]: Fixed in force push.

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by dsouzai — APPROVED** [review]

**Pull request merged**